client/src/app/payroll/payroll.component.html

```diff
@@ -1 +1,34 @@
-<div class="table-margin"><app-payroll-list></app-payroll-list></div>
+<div class="ml-3 mr-3">
+  <div class="row full-height">
+    <div class="col-2 background-dark border-light">
+      <div class="row">
+        <div class="list-group-item list-group-item-action">
+          <div class="row">
+            <div class="col-12">
+              <span>
+                Departments
+              </span>
+            </div>
+          </div>
+        </div>
+      </div>
+      <div
+        class="row"
+        *ngFor="let department of departments"
+        (click)="onSelectDepartment(department)"
```

Comment:
Although Angular makes it super easy to add a `(click)` binding to a `<div>`, it's generally a bad idea because `<div>` elements don't include any accessibility features for interactions like clicking/tapping.

If you want a whole 'card' or row to be clickable, then wrapping everything in an `<a href="#">...</a>` and styling it as a button is better.

> The a element can be wrapped around entire paragraphs, lists, tables, and so forth, even entire sections, so long as there is no interactive content within (e.g., buttons or other links). This example shows how this can be used to make an entire advertising block into a link:

[Source](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element)

Reply:
Also, while you're looking at HTML stuff, this is a good post about other HTML5 elements you can use to give your markup more meaning: https://dev.to/kenbellows/stop-using-so-many-divs-an-intro-to-semantic-html-3i9i

Reply:
Marking as "TODO" since we should come back to this. I have a lot to learn about html and accessibility.